ethcore/light/src/client/snapshot/chain.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! `RestorationTargetChain` impl for `HeaderChain`.
+
+
+use super::super::HeaderChain;
+use client::header_chain::PendingChanges;
+use ethcore::engines::EpochTransition;
+use ethcore::ids::BlockId;
+use ethcore::header::{Header, BlockNumber};
+use ethcore::snapshot::RestorationTargetChain;
+use ethereum_types::{H256, U256};
+use parking_lot::RwLock;
+use kvdb::DBTransaction;
+use ethcore::receipt::Receipt;
+use ethcore::views::BlockView;
+
+/// Wrapper for `HeaderChain` along with pending changes.
+pub struct LightChain {
+	chain: HeaderChain,
+	pending: RwLock<Option<PendingChanges>>
+}
+
+impl LightChain {
+	/// Create a wrapper for `HeaderChain` implementing `RestorationTargetChain`
+	pub fn new(chain: HeaderChain) -> Self {
+		LightChain {
+			chain: chain,
+			pending: RwLock::new(None)
+		}
+	}
+
+	/// Get a reference to the underlying chain
+	pub fn chain(&self) -> &HeaderChain {
+		&self.chain
+	}
+}
+
+impl RestorationTargetChain for LightChain {
+	fn genesis_hash(&self) -> H256 {
+		self.chain.genesis_hash()
+	}
+
+	fn genesis_header(&self) -> Header {
+		self.chain.genesis_header().decode().expect("genesis header is always decodable; qed")
+	}
+
+	fn block_hash(&self, index: BlockNumber) -> Option<H256> {
+		self.chain.block_hash(BlockId::Number(index))
+	}
+
+	fn block_header_data(&self, hash: &H256) -> Option<Header> {
+		self.chain.block_header(BlockId::Hash(hash.clone())).and_then(|h| h.decode().ok())
+	}
+
+	fn add_child(&self, _batch: &mut DBTransaction, _block_hash: H256, _child_hash: H256) {
+		// TODO
```

Comment:
Should we store this parent<->child relationship in the database similar to a full node 
https://github.com/paritytech/parity/blob/0bed5976e368e948566f5a445f5fa17088674f3d/ethcore/src/blockchain/blockchain.rs#L934-L947?

Reply:
yes, something like that